test/functional/p2p_fingerprint.py

```diff
@@ -101,34 +90,31 @@ def run_test(self):
 
         # Check that getdata request for stale block succeeds
         self.send_block_request(stale_hash, node0)
-        test_function = lambda: self.last_block_equals(stale_hash, node0)
-        self.wait_until(test_function, timeout=3)
+        node0.wait_for_block(stale_hash, timeout=3)
 
         # Check that getheader request for stale block header succeeds
         self.send_header_request(stale_hash, node0)
-        test_function = lambda: self.last_header_equals(stale_hash, node0)
-        self.wait_until(test_function, timeout=3)
+        node0.wait_for_header(hex(stale_hash), timeout=3)
 
         # Longest chain is extended so stale is much older than chain tip
         self.nodes[0].setmocktime(0)
-        tip = self.nodes[0].generatetoaddress(1, self.nodes[0].get_deterministic_priv_key().address)[0]
+        self.nodes[0].generatetoaddress(1, self.nodes[0].get_deterministic_priv_key().address)
         assert_equal(self.nodes[0].getblockcount(), 14)
-
-        # Send getdata & getheaders to refresh last received getheader message
-        block_hash = int(tip, 16)
-        self.send_block_request(block_hash, node0)
-        self.send_header_request(block_hash, node0)
-        node0.wait_for_block(block_hash, timeout=3)
+        node0.sync_with_ping()
```

Comment:
20047: should not undo 20466 here
```suggestion
        node0.wait_for_block(block_hash, timeout=3)
```